### DISCUSSION TITLE
IAM Admin User Removal

### DISCUSSION THREAD

**Discussion by samdammers** [cat-box/aws-foundry-ssl#10] [General]
Hey Team,

Nice work, I am currently having a look at this and doing a bit of my own automation.

I think however this oversteps its responsibilities where it creates as Admin IAM user.

You enter a realm of 'AWS Account Management' and I think there is potential for users to not fully understand and blitz past this with a weak password and not go back later to set MFA.
If people fail to read everything (Despite your warnings) they may not know they set a admin/Welcome1 administrator user for their entire AWS Account.

I would suggest removing entirely and linking off to AWS Best practice (https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#create-iam-users) at -> https://github.com/cat-box/aws-foundry-ssl/wiki/Before-You-Begin#creating-an-aws-account 

Or if you are still keen to include it, instead to make it an opt-in with warnings.

**Comment by DSBloom** [comment]
Where in the instructions do you create a password for this user? Is this the local admin credentials specified in the template parameters? I thought that was a user for the OS of the EC2 instance.

**Reply by cat-box** [reply]
This is the AdminUserPW field in cloudformation. I know it's a bit confusing, but the Admin user is a new account from which you can log into your aws console.

**Comment by cat-box** [comment]
While I see your concern about delving past foundry and into IAM user creation, we have automated the creation of an admin user for the purposes of enforcing the best practices you linked. 

I have noted your concern that MFA should be enforced and will update the wiki and template to emphasize its importance. However, just speaking from experience, if the end-user doesn't enable MFA after an automated account creation, they're likely not going to read through AWS' best practices to enable it either.

**Comment by samdammers** [comment]
That's fair, thanks for the response
